### PR TITLE
added newPillarBlock event to websocket

### PR DESCRIPTION
**Pull request by JakubFornadel**

### CODE REVIEW

File: libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp [hunk 1]
```diff
@@ -57,11 +62,59 @@ PillarBlock::Hash PillarBlock::getHash() {
...
+    res["validators_stakes_changes"].append(std::move(stake_change_json));
+  }
+  res["hash"] = dev::toJS(getHash());
+
+  return res;
+}
+
+dev::bytes PillarBlock::encodeSolidity() const {
+  dev::bytes result;
+  result.reserve((1 + 4 + 2 + 2 * validators_stakes_changes_.size()) * util::EncodingSolidity::kWordSize);
```
Comment (JakubFornadel) [inline libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp:90]: @kstdl what do these hardcoded values represent ?

File: libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp [hunk 2]
```diff
@@ -57,11 +62,59 @@ PillarBlock::Hash PillarBlock::getHash() {
...
+
+  auto start = util::EncodingSolidity::pack(32);
```
Comment (JakubFornadel) [inline libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp:92]: what is "start" for ?

File: tests/pillar_chain_test.cpp [hunk 3]
```diff
@@ -263,6 +264,171 @@ TEST_F(PillarChainTest, pillar_chain_syncing) {
...
+        "00000000000000000b00000000000000000000000000000000000000000000000000000000000000160000000000000000000000000000"
+        "000000000000000000000000000000000021000000000000000000000000000000000000000000000000000000000000002c0000000000"
+        "0000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000"
+        "000000050000000000000000000000000000000000000000000000000000000000000001ffffffffffffffffffffffffffffffffffffff"
+        "ffffffffffffffffffffffffff000000000000000000000000000000000000000000000000000000000000000200000000000000000000"
+        "000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000003ff"
+        "fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffd000000000000000000000000000000000000000000000000"
+        "00000000000000040000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000"
+        "0000000000000000000000000000000005fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffb");
```
Comment (JakubFornadel) [inline tests/pillar_chain_test.cpp:289]: how was this generated ?

File: libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp [hunk 4]
```diff
@@ -57,11 +62,59 @@ PillarBlock::Hash PillarBlock::getHash() {
...
+  result.insert(result.end(), start.begin(), start.end());
+
+  auto body = util::EncodingSolidity::pack(period_, state_root_, bridge_root_, previous_pillar_block_hash_);
+  result.insert(result.end(), body.begin(), body.end());
+
+  auto array_data = util::EncodingSolidity::pack(160, validators_stakes_changes_.size());
```
Comment (JakubFornadel) [inline libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp:98]: why 160 ?

File: libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp (hunk 2 above)
Comment (kstdl) [reply libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp:92]: this is a thing that you as always have for some reason in this encoding at the start

File: libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp (hunk 1 above)
Comment (kstdl) [reply libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp:90]: 1. So 1 is the start, 4 is count of variables. 2 should be for position and size of the dynamic data. It is better to move it ot separate variables to be cleaner 
2. Maybe I was doing something wrong there. But `stake change(s256) + address` is `2 * validators_stakes_changes_.size()`(all values are 32 bytes)

File: libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp (hunk 4 above)
Comment (kstdl) [reply libraries/core_libs/consensus/src/pillar_chain/pillar_block.cpp:98]: If I am not mistaken it is a position of start of the dynamic data (5*32)

**Pull request merged**